Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/PortalArtifactSystem.cs

```diff
@@ -0,0 +1,37 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Components;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Events;
+using Content.Shared.Mind;
+using Content.Shared.Storage;
+using Content.Shared.Teleportation.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Random;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Systems;
+
+public sealed class PortalArtifactSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly LinkedEntitySystem _link = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PortalArtifactComponent, ArtifactActivatedEvent>(OnActivate);
+    }
+
+    private void OnActivate(Entity<PortalArtifactComponent> artifact, ref ArtifactActivatedEvent args)
+    {
+        var firstPortal = Spawn(artifact.Comp.PortalProto, Transform(artifact.Owner).MapPosition);
```

Comment:
remove .Owner from both of these transform calls, the Entity itself casts to EntityUid automatically

Reply:
target its a MindContainerComponent, not Entity<>, so, owner is needed

Reply:
with proper refactor to EntityQueryEnumerator you get EntityUid directly